gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/QualityRuleServiceImpl.java

```diff
@@ -65,10 +65,17 @@ public class QualityRuleServiceImpl extends AbstractService implements QualityRu
     private AuditService auditService;
 
     @Override
-    public QualityRuleEntity findById(String id) {
+    public QualityRuleEntity findById(ExecutionContext executionContext, String id) {
```

Comment:
same thing, maybe only pass environmentId

Reply:
I will use `QualityRuleReferenceType referenceType, String referenceId` since a QualityRule has this `RefId/RefType` couple of fields.